### PR TITLE
Domain dependency

### PR DESCRIPTION
**Pull request by Guille88C**
This PR contains:

- Permissions library uses Either from domain library.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -4,6 +4,14 @@ allprojects {
     repositories {
         google()
         mavenCentral()
+
+        maven {
+            url = java.net.URI("https://maven.pkg.github.com/Rocket-Beer/rocket-core-data-network")
```
Comment (RuiMiguel) [inline build.gradle.kts]: why core-data-network??

File: core-data-permissions/src/main/java/com/rocket/android/core/data/permissions/PermissionError.kt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.rocket.android.core.data.permissions
 
-sealed class PermissionResponse {
-    data class SinglePermissionDenied(val data: PermissionDenied) : PermissionResponse()
+sealed class PermissionError {
```
Comment (RuiMiguel) [inline core-data-permissions/src/main/java/com/rocket/android/core/data/permissions/PermissionError.kt]: this pretend to be a Failure? must extend from Failure.FeatureFailure  from Domain

File: core-data-permissions/src/test/java/com/rocket/android/core/data/permissions/PermissionsTest.kt [hunk 3]
```diff
@@ -9,6 +9,7 @@ import com.rocket.android.core.data.mydexter.MyDexterResponse
 import com.rocket.android.core.data.mydexter.MyDexterSinglePermission
 import junit.framework.TestCase.assertEquals
 import junit.framework.TestCase.fail
+import kotlinx.coroutines.runBlocking
 import org.junit.Test
```
Comment (RuiMiguel) [inline core-data-permissions/src/test/java/com/rocket/android/core/data/permissions/PermissionsTest.kt:13]: use jUnit5 better

File: sample-app/src/main/java/com/rocket/android/core/data/sample/app/MainActivity.kt [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
 import android.os.Bundle
-import android.util.Log
-import com.rocket.android.core.data.permissions.Permissions
+import androidx.activity.viewModels
+import androidx.fragment.app.FragmentActivity
 import com.rocket.android.core.data.permissions.di.CoreDataProvider
 import com.rocket.android.core.data.sample.app.databinding.ActivityMainBinding
 
-class MainActivity : Activity() {
+class MainActivity : FragmentActivity() {
```
Comment (RuiMiguel) [inline sample-app/src/main/java/com/rocket/android/core/data/sample/app/MainActivity.kt:9]: WTF put all this into Compose right now!! 😂

**Review by RuiMiguel — CHANGES_REQUESTED** [review]

File: core-data-permissions/src/main/java/com/rocket/android/core/data/permissions/PermissionError.kt (hunk 2 above)
Comment (RuiMiguel) [reply core-data-permissions/src/main/java/com/rocket/android/core/data/permissions/PermissionError.kt]: maybe we need to change name for "FeatureFailure" to a better one...

File: build.gradle.kts (hunk 1 above)
Comment (Guille88C) [reply build.gradle.kts]: Good question xD, this should be domain 🤔 , I'll change it

File: core-data-permissions/src/main/java/com/rocket/android/core/data/permissions/PermissionError.kt (hunk 2 above)
Comment (Guille88C) [reply core-data-permissions/src/main/java/com/rocket/android/core/data/permissions/PermissionError.kt]: May be we should think in another name, but yeah, it should extend from Failure.FeatureFailure, I will change it!

File: core-data-permissions/src/test/java/com/rocket/android/core/data/permissions/PermissionsTest.kt (hunk 3 above)
Comment (Guille88C) [reply core-data-permissions/src/test/java/com/rocket/android/core/data/permissions/PermissionsTest.kt:13]: I will add a Task in the project to do it in another PR!

File: sample-app/src/main/java/com/rocket/android/core/data/sample/app/MainActivity.kt (hunk 4 above)
Comment (Guille88C) [reply sample-app/src/main/java/com/rocket/android/core/data/sample/app/MainActivity.kt:9]: OMG I should do it in compose 😢 , I will add a Task to the project!

File: core-data-permissions/src/test/java/com/rocket/android/core/data/permissions/PermissionsTest.kt (hunk 3 above)
Comment (Guille88C) [reply core-data-permissions/src/test/java/com/rocket/android/core/data/permissions/PermissionsTest.kt:13]: #5 issue created!

File: sample-app/src/main/java/com/rocket/android/core/data/sample/app/MainActivity.kt (hunk 4 above)
Comment (Guille88C) [reply sample-app/src/main/java/com/rocket/android/core/data/sample/app/MainActivity.kt:9]: #6 issue created!

**Review by RuiMiguel — APPROVED** [review]

**Pull request merged**